CMakeLists.txt

```diff
@@ -167,18 +167,20 @@ if(WIN32 AND ENABLE_NETSSL_WIN)
     list(APPEND Poco_COMPONENTS "NetSSL_Win")
 endif(WIN32 AND ENABLE_NETSSL_WIN)
 
-find_package(OpenSSL)
-if(OPENSSL_FOUND)
-    include_directories("${OPENSSL_INCLUDE_DIR}")
-    if(ENABLE_NETSSL)
-    add_subdirectory(NetSSL_OpenSSL)
-    list(APPEND Poco_COMPONENTS "NetSSL_OpenSSL")
-    endif()
-    if(ENABLE_CRYPTO)
-    add_subdirectory(Crypto)
-    list(APPEND Poco_COMPONENTS "Crypto")
-    endif()
-endif(OPENSSL_FOUND)
+if(ENABLE_NETSSL OR ENABLE_CRYPTO)
+    find_package(OpenSSL)
```

Comment:
I would mark this as required. So it fails. Otherwise it is confusing as it fails silently.